client/signup/steps/design-type-with-store/index.jsx

```diff
@@ -129,5 +83,72 @@ export default React.createClass( {
 
 		SignupActions.submitSignupStep( { stepName: this.props.stepName }, [], { designType } );
 		this.props.goToNextStep();
+	},
+
+	renderChoice( choice ) {
+		return (
+			<Card className="design-type-with-store__choice" key={ choice.type }>
+				<a className="design-type-with-store__choice__link" href="#" onClick={ ( event ) => this.handleChoiceClick( event, choice.type ) }>
+					{ choice.image }
+					<h2>{ choice.label }</h2>
+				</a>
+			</Card>
+		);
+	},
+
+	renderChoices() {
+		return (
+			<div className="design-type-with-store__list">
+				{ this.getChoices().map( this.renderChoice ) }
+			</div>
+		);
+	},
+
+	renderStoreStep() {
+		switch ( abtest( 'signupStoreBenchmarking' ) ) {
+			case 'bluehost':
+				return <BluehostStoreStep { ... this.props } onBackClick={ this.handleStoreBackClick } />;
+			case 'bluehostWithWoo':
+				return <BluehostStoreStep
+					{ ... this.props }
+					onBackClick={ this.handleStoreBackClick }
+					partnerName='Bluehost with WooCommerce'
+					partnerUrl='https://www.bluehost.com/web-hosting/signup?flow=woocommerce'
+				/>;
+			case 'siteground':
+				return <SitegroundStoreStep { ... this.props } onBackClick={ this.handleStoreBackClick } />;
+			default:
+				return <PressableStoreStep { ... this.props } ref={ this.onPressableStoreStepRef } onBackClick={ this.handleStoreBackClick } />;
+		}
+	},
+
+	render() {
+		const storeWrapperClassName = classNames( {
+			'design-type-with-store__store-wrapper': true,
+			'is-hidden': ! this.state.showStore,
+		} );
+
+		const sectionWrapperClassName = classNames( {
+			'design-type-with-store__section-wrapper': true,
+			'is-hidden': this.state.showStore,
+		} );
+
+		return (
+			<div className="design-type-with-store">
+				<div className={ storeWrapperClassName } >
+					{ this.renderStoreStep() }
+				</div>
+				<div className={ sectionWrapperClassName }>
+					<StepWrapper
+						flowName={ this.props.flowName }
+						stepName={ this.props.stepName }
+						positionInFlow={ this.props.positionInFlow }
+						fallbackHeaderText={ this.translate( 'What would you like your homepage to look like?' ) }
+						fallbackSubHeaderText={ this.translate( 'This will help us figure out what kinds of designs to show you.' ) }
+						signupProgressStore={ this.props.signupProgressStore }
+						stepContent={ this.renderChoices() } />
```

Comment:
It was the way we have things pulled out into two separate `renderStuff()` functions that led me here, wondering why we couldn't simply inline them here. Typically we prefer to have things explicit in `render()` unless it's very complicated (and then probably we ought to create new components instead). Then I saw that we're passing this stuff in as a `prop` to the `<StepWrapper />` which is a very strange way of doing things in my opinion.

All this is to say that while maybe this is okay for this PR, I think it would be great to followup with a quick PR to adjust the `<StepWrapper />` to use `children`, which are just what it probably should be using.

```js
<StepWrapper
	flowName={ … }
	…
>
	<div className="design-type-with-store__list">
		{ this.getChoices().map( choice => (
			<Card className="design-type-with-store__choice" key={ choice.type }>
				<a className="design-type-with-store__choice__link" href="#" onClick={ ( event ) => this.handleChoiceClick( event, choice.type ) }>
					{ choice.image }
					<h2>{ choice.label }</h2>
				</a>
			</Card>
		 ) ) }
 	</div>
</StepWrapper>
```

and then in `<StepWrapper />` instead of using `{ stepContent }` we would insert `{ this.props.children }`